pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -5434,3 +5434,129 @@ func TestClusterLabelsRHPAMDashbuilderDefaultEnvironment(t *testing.T) {
+				},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.True(t, len(cr.Spec.Objects.Servers[0].Kafka.Topics) == 2)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	autocreateEnv, maxBlockMS, topics, extEnabled, clientID, acks, groupID, bootstrapServers := false, false, false, false, false, false, false, false
+	for _, env := range envs {
+
+		if env.Name == "KIE_SERVER_KAFKA_EXT_ENABLED" {
+			extEnabled = true
+			assert.True(t, env.Value == "true")
+		}
+
+		if env.Name == constants.KieServerKafkaExtGroupID {
+			groupID = true
+			assert.True(t, env.Value == "my-kafka-group")
+		}
+
+		if env.Name == constants.KieServerKafkaExtAcks {
+			acks = true
+			assert.True(t, env.Value == "2")
+		}
+
+		if env.Name == constants.KieServerKafkaExtAutocreateTopics {
+			autocreateEnv = true
+			assert.True(t, env.Value == "true")
+		}
+
+		if env.Name == constants.KieServerKafkaExtMaxBlockMS {
+			maxBlockMS = true
+			assert.True(t, env.Value == "2100")
+		}
+
+		if env.Name == constants.KieServerKafkaExtClientID {
+			clientID = true
+			assert.True(t, env.Value == "C1234567")
+		}
+
+		if env.Name == constants.KieServerKafkaExtServers {
+			bootstrapServers = true
+			assert.True(t, env.Value == "localhost:9092")
+		}
+
+		if strings.HasPrefix(env.Name, constants.KieServerKafkaExtTopics) {
+			topics = true
+			assert.True(t, env.Value == "[{events=my-topics} {errors=my-errs}]")
+		}
+	}
+
+	assert.True(t, autocreateEnv)
+	assert.True(t, maxBlockMS)
+	assert.True(t, topics)
+	assert.True(t, extEnabled)
+	assert.True(t, clientID)
+	assert.True(t, acks)
+	assert.True(t, groupID)
+	assert.True(t, bootstrapServers)
+}
+
+func createKafkaExtObject() *api.KafkaExtObject {
+	kafkaExtObject := api.KafkaExtObject{
+		MaxBlockMs:       2100,
+		AutocreateTopics: Pbool(true),
+		BootstrapServers: "localhost:9092",
+		GroupID:          "my-kafka-group",
+		Acks:             2,
+		Topics: []api.Topic{
+			{Name: "events=my-topics"},
+			{Name: "errors=my-errs"},
+		},
+		ExtEnabled: true,
+		ClientID:   "C1234567",
+	}
+	return &kafkaExtObject
+}
+
+func TestRhdmEnvironmentWithKafkaExtDefault(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhdmProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{
+					{
+						Kafka: &api.KafkaExtObject{},
+					},
+				},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	extEnabled := true
+
+	for _, env := range envs {
+
+		if env.Name == "KIE_SERVER_KAFKA_EXT_ENABLED" {
+			extEnabled = true
+			assert.True(t, env.Value == "false")
+		}
+	}
+	assert.True(t, extEnabled)
+}
```

Comment:
we need to add a check somewhere for what happens when the kafka object is empty. the object should be empty in spec and status, and the env vars should not exist.